tests/gmap_test.py

```diff
@@ -57,12 +77,30 @@ def f(x):
 
     x = jnp.arange(800).reshape((8, 10, 10))
 
-    for loop, n in schedule:
-      approx_n = x.shape[0] if n is None else n
-      if loop == 'parallel' and approx_n > xla_bridge.device_count():
-        raise SkipTest("this test requires more XLA devices")
+    self.assertAllClose(gmap(f, schedule)(x), vmap(f)(x))
 
-    self.assertAllClose(vmap(f)(x), gmap(f, schedule)(x))
+  @check_default_schedules(lambda s: not any(c[0] == 'sequential' for c in s))
+  @skip_insufficient_devices(8)
+  @ignore_gmap_warning()
+  @skipIf(not config.omnistaging_enabled,
+          "vmap collectives only supported when omnistaging is enabled")
+  def testAxisName(self, schedule):
+    def f(x):
+      return x - lax.psum(x, 'i')
+    x = jnp.arange(8)
+    self.assertAllClose(gmap(f, schedule, axis_name='i')(x),
+                        vmap(f, axis_name='i')(x))
+
+  @ignore_gmap_warning()
+  @skipIf(not config.omnistaging_enabled,
```

Comment:
Just to double-check, are you able to run these tests by setting the omnistaging env var or similar? I ask because if flag parsing happens after module import time (i.e. after this expression is evaluated) it might not be possible to enable these tests.

Reply:
yeah those tests run if I prefix my `pytest` invocation with `JAX_OMNISTAGING=1`